src/compiler/checker.ts

```diff
@@ -56,6 +56,10 @@ namespace ts {
 
         const undefinedSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "undefined");
         undefinedSymbol.declarations = [];
+        const NaNSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.TypeAlias | SymbolFlags.Transient, (NaN).toString());

```

Comment:
Possibly. What flag would be correct for a type merged with a value? Alias simply yielded correct results in output (and makes some amount of sense - the value aliases the type when used in a type position)